lib/nerves_hub_web/channels/device_channel.ex

```diff
@@ -261,10 +261,20 @@ defmodule NervesHubWeb.DeviceChannel do
     end
   end
 
-  # TODO this is wrong
   def handle_info(%Broadcast{event: "moved"}, socket) do
     device = Repo.reload(socket.assigns.device)
-    Presence.update(device, %{product_id: device.product_id})
+
+    Registry.update_value(NervesHub.Devices, device.id, fn _ ->
+      %{deployment_id: device.deployment_id}
+    end)
+
+    Presence.update(device, %{
+      product_id: device.product_id,
+      deployment_id: device.deployment_id
+    })
+
+    # The old deployment is no longer valid, so let's look one up again
+    send(self(), :resolve_changed_deployment)
```

Comment:
```suggestion
    handle_info(:resolve_changed_deployment, assign(socket, device: device))
```

We could save a message round trip here by just handling it directly. Not crucial though, just a thought

Reply:
I merged it since I tested what I had. I don't think this will be a performance problem and it feels better to me to keep it separate, but that's purely a feeling lol